liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -313,7 +313,11 @@ public boolean recompileBuildFile(File buildFile, List<String> artifactPaths, Th
                     List<String> dependencyIds = new ArrayList<String>();
                     List<Artifact> updatedArtifacts = getNewDependencies(dependencies, this.existingDependencies);
 
-                    if (!updatedArtifacts.isEmpty()) {
+                    // handle the case where an invalid dependency is added
+                    if (updatedArtifacts.isEmpty() && (this.existingDependencies.size() != dependencies.size())) {
+                        // make sure lists match so user does not see the invalid dependency error more than once
```

Comment:
I added a [comment](https://github.com/OpenLiberty/ci.maven/issues/595#issuecomment-532803345) in the issue, it should still show the error in this case if the invalid dependency is still in the pom.

The problem now is, if you add two dependencies at the same time, one which is invalid and the second which is valid, line 364 which does `this.existingDependencies = dependencies;` would make it stop giving an error about the invalid dependency when you make further updates to the pom.  But we want it to always show the error about the invalid one. 

A possible solution could be to always call `getArtifact()` on the ENTIRE list of dependencies from the pom, so it will give an error for any that DON'T exist.  And `existingDependencies` should always be updated to be the list of dependencies that DO exist.

Reply:
The issue I mentioned above still exists.
e.g.
1. Add an invalid dependency - a warning is shown (which is correct).
2. Add a valid dependency in addition to the invalid one - a warning is still shown for the invalid one (which is correct).
3. Add another valid dependency - no more warnings are shown, even though the invalid dependency in 1. is still in the pom.

Reply:
Addressed with latest commit https://github.com/OpenLiberty/ci.maven/pull/604/commits/0c4c9c8fb2af4f2373b0d2def8433d390ee15f5a